### DISCUSSION TITLE
Exploring Alternative OTP Service Providers: Configuring for MSG91

### DISCUSSION THREAD

**Discussion by dhwani-ankit** [egovernments/DIGIT-DevOps#1826] [Q&A] [answered]
Hey, I have a query regarding the usage of an OTP service other than the one mentioned in the configuration file at https://services.mgov.gov.in/. If we want to use a different OTP service, such as MSG91, where should I make the necessary changes or updates?

**Comment by talele08** [comment, marked as answer]
All SMS in DIGIT are sent by egov-notification-sms service. It is basically a kafka consumer which consumes messages from the notification topic and then calls the third part API provider to send the SMS. OTP is sent to user by calling the _send API of user-otp service. The user-otp service will generate the otp and the text message and publish it on notification topic. egov-notification-sms service will consume that message and call the third party API to send the otp. If you want to use different SMS provider, please do the integration in egov-notification-sms

**Reply by dhwani-ankit** [reply]
We integrated custom code in the user-otp service to send OTPs using MSG91. But, we're facing a "BadRequest" also "No username and password found" error while trying to send OTPs. We have made some database updates to the phone numbers and added new entries.

**Reply by shashwat-egov** [reply]
Hi, @dhwani-ankit , the custom implementation of MSG91 should be there in egov-sms-notification service as opposed to user-otp service as user-otp merely generates SMS request and publishes it on kafka for the egov-sms-notification service to consume.

**Comment by dhwani-ankit** [comment]
Also Adding one more query:

We are currently working on adding our custom module to egov-workflow-v2. However, when attempting to deploy our custom image, we encounter an error stating that we don't have access to pull the image with the name: egov-workflow-v2-db.

**Reply by nikhilmulinti** [reply]
@dhwani-ankit You are encountering an access denied error because you're trying to pull a Docker image from a private registry. To resolve this, please create a Kubernetes secret based on your Docker Hub credentials and reference that secret in your application deployment. For a detailed configuration, refer to this document: https://kubernetes.io/docs/tasks/configure-pod-container/pull-image-private-registry/.

**Comment by dhwani-ankit** [comment]
Also Adding one more query:

We integrated custom code in the user-otp service to send OTPs using MSG91. But, we're facing a "BadRequest" also "No username and password found" error while trying to send OTPs. We have made some database updates to the phone numbers and added new entries.
